File: deno/send_message/src/functions/send_message_discord_webhook.ts
export default async function (variables: any, message: string) {
  const webhook = variables["DISCORD_WEBHOOK_URL"];

  if (!webhook) {
    return {
      success: false,
      error: "DISCORD_WEBHOOK_URL is not set",
    };
  }

  const res = await fetch(webhook, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      content: message,
    }),
  });
  const data = await res.json();
  if (res.status !== 204) {
    return {
      success: false,
      error: data.message,
    };
  }

  return { success: true };
}

File: python/send_message/functions/send_message_discord_webhook.py
import json
import requests


def send_message_discord_webhook(variables, message):
    """Send message to Discord webhook"""

    if not message:
        raise Exception("No message provided")

    webhook = variables.get("DISCORD_WEBHOOK_URL", None)

    if not webhook:
        raise Exception("No Discord webhook URL provided")

    try:
        response = requests.post(
            webhook,
            data=json.dumps({"content": message}),
            headers={"Content-Type": "application/json"},
        )
        response.raise_for_status()
    except Exception as e:
        print(e)
        return {"success": False, "error": str(e)}

    return {"success": True}
